### PR TITLE
removing updated value and takes default value

### PR DESCRIPTION
**Pull request by amolpati30**
Added to the entities with their respective locators:

- `update_variable_value`: Updating the ansible variable value in the host.
- `del_variable_value`: Deletes the updated value, reverts to the default value.
- `read_variable_value`: Read the value of ansible variable in the host.

Dependent PR: https://github.com/SatelliteQE/robottelo/pull/16969

### CODE REVIEW

**Comment by amolpati30** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_ansible.py::TestAnsibleCfgMgmt -k test_positive_assign_ansible_role_variable_on_host
robottelo: 16969

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 460
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_ansible.py::TestAnsibleCfgMgmt -k test_positive_assign_ansible_role_variable_on_host --external-logging
Test Result : ========== 1 passed, 10 deselected, 24 warnings in 837.50s (0:13:57) ===========
```

File: airgun/entities/host_new.py [hunk 1]
```diff
@@ -975,6 +975,22 @@ def refresh_applicability(self, entity_name):
         self.browser.plugin.ensure_page_safe()
         view.dropdown.item_select('Refresh applicability')
 
+    def read_update_value(self, entity_name, key, value):
```
Comment (LadislavVasina1) [inline airgun/entities/host_new.py:978]: Could you please change the wording of the function name.
```suggestion
    def read_updated_value(self, entity_name, key, value):
```

File: airgun/entities/host_new.py [hunk 2]
```diff
@@ -975,6 +975,22 @@ def refresh_applicability(self, entity_name):
...
+        view = self.navigate_to(self, 'NewDetails', entity_name=entity_name)
+        wait_for(lambda: view.ansible.variables.table.is_displayed, timeout=10)
+        view.ansible.variables.table.row(name=key)[5].widget.click()
```
Comment (LadislavVasina1) [inline airgun/entities/host_new.py:981]: Can you please add some comment to the `5` index

**Review by LadislavVasina1 — CHANGES_REQUESTED** [review]
Small change requested, otherwise looks good to me. GJ @amolpati30!

File: airgun/entities/host_new.py (hunk 1 above)
Comment (amolpati30) [reply airgun/entities/host_new.py]: Updated the function name.

File: airgun/entities/host_new.py (hunk 2 above)
Comment (amolpati30) [reply airgun/entities/host_new.py:982]: Added comment for the [5] index.

**Comment by amolpati30** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_ansible.py::TestAnsibleCfgMgmt -k test_positive_assign_ansible_role_variable_on_host
robottelo: 16969

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 461
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_ansible.py::TestAnsibleCfgMgmt -k test_positive_assign_ansible_role_variable_on_host --external-logging
Test Result : ========== 1 failed, 10 deselected, 22 warnings in 800.28s (0:13:20) ===========
```

**Comment by amolpati30** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_ansible.py::TestAnsibleCfgMgmt -k test_positive_assign_ansible_role_variable_on_host
robottelo: 16969

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 462
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_ansible.py::TestAnsibleCfgMgmt -k test_positive_assign_ansible_role_variable_on_host --external-logging
Test Result : ========== 1 passed, 10 deselected, 24 warnings in 892.06s (0:14:52) ===========
```

**Review by LadislavVasina1 — APPROVED** [review]
ACK

**Comment by lhellebr** [comment]
Is there a reason why each of these functions isn't an atomic function? Is it important to read RIGHT AFTER updating/deleting?

**Comment by amolpati30** [comment]
@lhellebr 
While working on this entity, I noticed that some features do not have a specific column for each actions item in the table. For example, actions like Submit, Cancel, and Edit point to a single column (indexed as 5). However, when we update a value and attempt to delete it, a new "3-dot" option appears, which does not correspond to any specific column, even though it is part of the table.
Due to these differences, two distinct methods are required. And I am reading the values here for assertion purposes, to verify the data after update and delete.

**Comment by lhellebr** [comment]
Let me rephrase my question. Let's take one of the functions: `read_value_after_del`. Why is it a single function? What is so specific about operation "delete and then read"? I would think this can be two functions: `delete` and `read` and the test can just call one after the other.

**Comment by amolpati30** [comment]
Thank you for the clarification. I would suggest that if we are deleting an updated value, we should read the value for verification/assertion purposes. I have already added this functionality in the method. However, creating a separate method specifically for reading the value is also a good idea.

This way, we could have two methods for update/delete operations and a third method to handle reading the value. Do you think I should update the implementation accordingly?

**Comment by lhellebr** [comment]
I think yes, if you don't have a specific reason to make those inherently atomic operations a single operation in this case. I think that's also the way we have it everywhere else.

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 463
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_ansible.py::TestAnsibleCfgMgmt -k test_positive_assign_ansible_role_variable_on_host --external-logging
Test Result : ========== 1 failed, 10 deselected, 26 warnings in 769.00s (0:12:49) ===========
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 464
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_ansible.py::TestAnsibleCfgMgmt -k test_positive_assign_ansible_role_variable_on_host --external-logging
Test Result : ========== 1 passed, 10 deselected, 26 warnings in 1420.64s (0:23:40) ==========
```

**Review by LadislavVasina1 — APPROVED** [review]

File: airgun/views/host_new.py [hunk 3]
```diff
@@ -513,20 +513,30 @@ class roles(Tab):
...
+                    'Value': TextInput(locator='//textarea[contains(@class, "pf-c-form")]'),
                     'Source attribute': Text('./span'),
                     # The next 2 buttons are hidden by default, but appear in this order
-                    5: Button(locator='.//button[@aria-label="Cancel editing override button"]'),
-                    6: Button(locator='.//button[@aria-label="Submit override button"]'),
+                    6: Button(locator='.//button[@aria-label="Cancel editing override button"]'),
+                    7: Button(locator='.//button[@aria-label="Submit override button"]'),
                     # Clicking this button hides it, and displays the previous 2
-                    7: Button(locator='.//button[@aria-label="Edit override button"]'),
+                    5: Button(locator='.//button[@aria-label="Edit override button"]'),
```
Comment (Gauravtalreja1) [inline airgun/views/host_new.py:533]: Why changing the buttons order here? 
and could you add some descriptions about change to this locators in the description?

Comment (amolpati30) [reply airgun/views/host_new.py:533]: Locators 6 and 7 are not supported, and there is no column for the edit, submit, or cancel buttons. Based on the index, these buttons correspond to key 5. To enable editing, I added the edit locator under key 5. For submitting the value, I created another variable with the same key (5) but for the submit action. However, combining them does not work—I tried that as well.

**Review by Gauravtalreja1 — APPROVED** [review]
